micronota/bfillings/diamond.py

```diff
@@ -325,68 +335,76 @@ def parse_tabular(diamond_res, column='bitscore'):
-            The column used to pick the best hits.
-
         Returns
         -------
         pandas.DataFrame
             The best matched records for each query sequence.
         '''
         seqs = read(diamond_res, format='sam')
-        columns = ['qseqid', 'sseqid', 'pident', 'length', 'mismatch',
-                   'gapopen', 'qstart', 'qend', 'sstart', 'send',
-                   'evalue', 'bitscore', 'sequence']
+        columns = ['qseqid', 'sseqid', 'pident', 'qlen', 'mismatch',
+                   'qstart', 'sstart', 'evalue', 'bitscore', 'sseq']
         df = pd.DataFrame(columns=columns)
         for i, seq in enumerate(seqs):
-            s = str(seq)
+            sseq = str(seq)
 
             qseqid = seq.metadata['QNAME']
             sseqid = seq.metadata['RNAME']
             pident = seq.metadata['ZI']
-            length = seq.metadata['ZL']
+            qlen = seq.metadata['ZL']
             mismatch = seq.metadata['CIGAR']
-            gapopen = ''
             qstart = seq.metadata['POS']
-            qend = ''
             sstart = seq.metadata['ZS']
-            send = ''
             evalue = seq.metadata['ZE']
             bitscore = seq.metadata['ZR']
             row = pd.Series([qseqid, sseqid, pident,
-                             length, mismatch, gapopen,
-                             qstart, qend, sstart, send,
-                             evalue, bitscore, s],
+                             qlen, mismatch,
+                             qstart, sstart,
+                             evalue, bitscore, sseq],
                             index=columns)
             df.loc[i] = row
 
-        if column is not None:
-            idx = df.groupby('qseqid')[column].idxmax()
-            df_max = df.loc[idx]
-            df_max.index = idx.index
-            df = df_max[['sseqid', 'evalue', 'bitscore', 'sequence']]
-        else:
-            df = df[['sseqid', 'evalue', 'bitscore', 'sequence']]
         return df
 
+    @staticmethod
+    def _filter_best(df, column='evalue'):
+        # pick the rows that have highest bitscore for each qseqid
+        # df_max = df.groupby('qseqid').apply(
+        #     lambda r: r[r[column] == r[column].max()])
+        if column == 'evalue':
+            idx = df.groupby('qseqid')[column].idxmin()
+        elif column == 'bitscore':
+            idx = df.groupby('qseqid')[column].idxmax()
+        df_best = df.loc[idx]
+        df_best.index = idx.index
+        return df_best
+
+    @staticmethod
+    def _filter_id_cov(df, pident=90, cov=80):
+        '''Filter away the hits using the same UniRef clustering standards.'''
+        select_id = df.pident >= pident
+        aligned_length = df.mismatch.apply(_compute_aligned_length)
+        select_cov = ((aligned_length * 100 / df.qlen >= cov) &
+                      (aligned_length * 100 / df.sseq.apply(len) >= cov))
+        # if qlen * 100 / len(row.sequence) >= 80:
+        return df[select_id & select_cov]
```

Comment:
This is the same implementation that I was thinking.

At the moment it isn't being used.  Probably will want to call this within parse_sam, correct?